### PR TITLE
Exit if pre-up hook exits non-zero

### PR DESCRIPTION
**Pull request by pbrisbin**

### CODE REVIEW

File: share/rcm.sh.in [hunk 1]
```diff
@@ -143,6 +143,12 @@ run_hooks() {
         find "$hook_file" -type f \( \( -user $LOGNAME -perm -100 \) -o -perm -001 \) \
           | sort | while read file; do
             sh -c 'cd -- "`dirname $1`" && ./"`basename $1`"' arg0 "$file"
+            hook_ret=$?
+
+            if [ "$hook_ret" -ne 0 ]; then
+              echo "$when-$direction hook exited non-zero ($hook_ret)" >&2
```
Comment (pbrisbin) [inline share/rcm.sh.in:149]: Should probably include the file here.

```suggestion
              echo "$when-$direction hook $file exited non-zero ($hook_ret)" >&2
```

Perhaps `$when-$direction` isn't needed then?

File: bin/rcup.in [hunk 2]
```diff
@@ -296,6 +296,11 @@ handle_command_line "$@"
 : ${DOTFILES_DIRS:=$DOTFILES_DIRS $DEFAULT_DOTFILES_DIR}
 
 run_hooks pre up
+pre_up_ret=$?
+
+if [ "$pre_up_ret" -ne 0 ]; then
```
Comment (pbrisbin) [inline bin/rcup.in:301]: Pretty easy to guard this by a CLI flag if desired, I think.

File: share/rcm.sh.in [hunk 3]
```diff
@@ -143,6 +143,12 @@ run_hooks() {
...
+              exit $hook_ret # NB. this only exits the while-read; caller must still check and exit
```
Comment (mike-burns) [inline share/rcm.sh.in:150]: Whoa, that's good to know. Normally I don't love code comments but I definitely would have missed that quirk.

File: share/rcm.sh.in (hunk 1 above)
Comment (mike-burns) [reply share/rcm.sh.in:149]: +1 to adding `$file`, and I think more information is better in the case of this failure. Save some debugging effort: leave the `$when-$direction`.

**Review by mike-burns — COMMENTED** [review]
LGTM, actually. Thanks for catching this.

**Comment by pbrisbin** [comment]
Sweet. Thanks for the quick review! Should be good to merge now.

**Review by mike-burns — APPROVED** [review]
Thanks for this. I can merge it Friday, and see if I can get a release out this year. 'Tis the season!

**Comment by mike-burns** [comment]
Thanks. Also adding one for the down case.

**Pull request merged**